test/server_test.py

```diff
@@ -14,55 +14,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
+import os
 import multiprocessing
 import random
 import time
-from helpers import unittest
-try:
-    from urllib2 import Request, urlopen, HTTPError
-except ImportError:
-    from urllib.request import Request, urlopen
-    from urllib.error import HTTPError
 
+from helpers import unittest, with_config
+import luigi.rpc
 import luigi.server
+from luigi.scheduler import CentralPlannerScheduler
 
+from tornado.testing import AsyncHTTPTestCase
 
-def run_server(api_port):
-    luigi.server.run(api_port=api_port, address='127.0.0.1')
 
+class ServerTestBase(AsyncHTTPTestCase):
 
-class ServerTestBase(unittest.TestCase):
+    def get_app(self):
+        return luigi.server.app(CentralPlannerScheduler())
 
     def setUp(self):
-        self._api_port = random.randint(1024, 9999)
-        self._process = multiprocessing.Process(target=run_server, args=(self._api_port,))
-        self._process.start()
-        time.sleep(0.1)  # wait for server to start
+        super(ServerTestBase, self).setUp()
+
+        self._old_fetch = luigi.rpc.RemoteScheduler._fetch
+
+        def _fetch(obj, url, body, *args, **kwargs):
+            response = self.fetch(url, body=body, method='POST')
+            if response.code >= 400:
+                raise luigi.rpc.RPCError(
+                    'Errror when connecting to remote scheduler'
+                )
+            return response.body.decode('utf-8')
+
+        luigi.rpc.RemoteScheduler._fetch = _fetch
```

Comment:
Using the AsyncHTTPClient should work. It's what do `self.fecth`. It just requires an IOLoop running. The client automatically start one if there isn't. I thought about it, but I thought it would be relatively complex, but it seems simple. I can try to modify RemoteScheduler to use AsyncHTTPClient.